src/MarginTrading.AssetService.Contracts/Extensions/TickFormulaExtensions.cs

```diff
@@ -37,7 +37,8 @@ public static class TickFormulaExtensions
         
         public static uint GetPrecision(this decimal number)
         {
-            return BitConverter.GetBytes(decimal.GetBits(number)[3])[2];
+            var g29 = decimal.Parse(number.ToString("G29"));
```

Comment:
I'm curious why the number argument is needed to be converted into a string with G29 format.

As far as I understand from the ticket, this development is needed to ignore the values after the comma for integer values. (e.g. 1234.0 -> 1234).

It can be achievable by checking if  `num % 1 == 0`. It will evaluate **true** if `num` is an integer so that we can return 0 from this function.  And the existing code that checks the last 32 bits can keep running as it used to. I'm afraid converting the number into string and parsing again can compromise performance and can have possible side effects.

Reply:
From what I understand from the ticket, the stakeholder only complained about **integers** -the numbers having a precision of '0'-. The values (1234.1000,1234.1100 are not a problem for them.). 

The suggested change should resolve the problem in the ticket description.

```suggestion
             if(number % 1 == 0)
                return 0;

            return BitConverter.GetBytes(decimal.GetBits(number)[3])[2];
```

Please let me know if I'm missing something, but I believe converting a decimal into a string and parsing it again must be the last resort.

Reply:
yes, but in this case
1.0 = 0
1.10 = 2
1.110 = 3
...

we can have problems with other precisions in the future, no?
@tarurar , any suggestions?